app/models/concerns/orchestration.rb

```diff
@@ -162,7 +162,13 @@ def process(queue_name)
     # if we have no failures - we are done
     return true if q.failed.empty? && q.pending.empty? && q.conflict.empty? && orchestration_errors?
 
-    logger.warn "Rolling back due to a problem: #{q.failed + q.conflict}"
+    if !q.failed.empty? || !q.conflict.empty?
+      logger.warn "Rolling back due to a problem: #{q.failed + q.conflict}"
+    elsif !q.pending.empty?
+      logger.warn "Rolling back due to unexpected pending task(s): #{q.pending}"
+    else
+      logger.warn "Rolling back due to errors: #{errors}"
```

Comment:
Any thoughts on merging in the return so it's easier to reduce duplication? I think it'd make it easier to follow what's going on:
```suggestion
    elsif orchestration_errors?
      logger.warn "Rolling back due to errors: #{errors}"
    else
      # if we have no failures - we are done
      return true
```